### PR TITLE
Improve update tool and documentation

### PR DESCRIPTION
**Pull request by SupunS**
## Description

Improves the update tool to:
- Pretty print the outputs
- Make the protocol of GitHub API configurable (HTTPS vs SSH).
- Improve documentation
______

- [x] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

File: tools/update/main.ts [hunk 1]
```diff
@@ -331,7 +336,7 @@ class Updater {
         const dir = await mkdtemp(path.join(os.tmpdir(), `${owner}-${repoName}`))
 
         console.log(`Cloning ${fullRepoName} ...`)
-        await exec(`git clone git@github.com:${fullRepoName} ${dir}`)
```
Comment (SupunS) [inline tools/update/main.ts:334]: This assumes GitHub ssh is pre-configured. Switched to the HTTPS API for simplicity.

File: tools/update/main.ts [hunk 2]
```diff
@@ -463,7 +468,7 @@ class Releaser {
         const dir = await mkdtemp(path.join(os.tmpdir(), `${owner}-${repoName}`))
 
         console.log(`Cloning ${this.repo} ...`)
-        await exec(`git clone git@github.com:${this.repo} ${dir}`)
+        await exec(`git clone https://github.com/${this.repo} ${dir}`)
```
Comment (turbolent) [inline tools/update/main.ts:471]: This

File: tools/update/main.ts (hunk 1 above)
Comment (turbolent) [reply tools/update/main.ts:334]: I see the problem, though hard-coding HTTPS assumes that that is configured, but users usually only have one or the other configured. Maybe make this a CLI option? Happy that it defaults to HTTPS, I guess that is more common?

File: tools/update/main.ts [hunk 3]
```diff
@@ -560,3 +565,10 @@ class Releaser {
         )
         .parse()
 })()
+
+async function runWithConsoleGroup(func: () => Promise<boolean>): Promise<boolean> {
+    console.group()
+    const result = func()
+    result.finally(console.groupEnd)
```
Comment (turbolent) [inline tools/update/main.ts:572]: 😍 

Had no idea Node had this feature, neat!

File: tools/update/main.ts [hunk 4]
```diff
@@ -159,15 +162,15 @@ class Updater {
...
+        console.log(`> Checking if all mods of repo ${fullRepoName} at version ${refName} are updated ...`)
 
         for (const mod of repo.mods) {
             if (!await this.repoModUpdated(refName, repo, mod)) {
                 return false
             }
         }
 
-        console.log(`All mods of mod ${fullRepoName} at repo version ${refName} are up-to-date`)
+        console.log(`✓ All mods of mod ${fullRepoName} at repo version ${refName} are up-to-date`)
```
Comment (turbolent) [inline tools/update/main.ts:173]: 👌

**Review by turbolent — APPROVED** [review]
Nice! 

Thank you for improving the documentation and the output, so much better 👌

File: tools/update/main.ts (hunk 1 above)
Comment (SupunS) [reply tools/update/main.ts:334]: hmm yeah, making it configurable sounds good!

Comment (SupunS) [reply tools/update/main.ts:334]: made it configurable: 6faf6765a0c6a818be02729cc7108f9aef637138

**Review by turbolent — APPROVED** [review]

**Pull request merged**